File: js/nova/math/Sequence.js
var Sequence = function () {
	this.values = nova_null;
	
	
};

Sequence.prototype.Sequence = function (values) {
	this.this(values);
};

Sequence.prototype.destroy = function () {
};

Sequence.prototype.this = function (values) {
	this = values;
};

Sequence.prototype.sum = function (num) {
	var value;
	var i;
	value = 0;
	i = 0;
	for (; i < num; i++) {
		value = value + this.values.get(i);
	}
	return value;
};

Sequence.prototype.super = function () {
};


(function () {
		this.INFINITE = -1;
})();

File: java/nova/math/Sequence.java
package nova.math;

import java.util.Optional;
import nova.exception.ExceptionData;
import nova.exception.Exception;
import nova.exception.DivideByZeroException;
import nova.io.NovaConsole;
import nova.primitive.number.Number;
import nova.primitive.number.NovaByte;
import nova.primitive.number.NovaShort;
import nova.primitive.number.NovaInt;
import nova.primitive.number.NovaLong;
import nova.primitive.number.NovaFloat;
import nova.primitive.number.NovaDouble;
import nova.primitive.Null;
import nova.primitive.number.Char;
import nova.primitive.Bool;
import nova.datastruct.list.NovaArray;
import nova.datastruct.list.IntArray;
import nova.datastruct.list.CharArray;
import nova.datastruct.list.DoubleArray;
import nova.datastruct.list.IntRange;
import nova.thread.Thread;
import nova.thread.async.Async;
import nova.gc.GC;
import nova.math.NovaMath;
import nova.NovaObject;
import nova.NovaString;
import nova.System;
import nova.Class;

public class Sequence extends NovaObject
{
	
	public DoubleArray values;
	
	
	public static int INFINITE;
	
	
	public Sequence(DoubleArray values)
	{
		init(values);
	}
	
	public void init(DoubleArray values)
	{
		this.values = values;
	}
	
	public static double sum(int num)
	{
		double value;
		int i;
		value = 0;
		i = (int)0;
		for (; i < (int)num; i++)
		{
			value = value + values.get(i);
		}
		return value;
	}
	
}
